### PR TITLE
Fix bug where nav doesn't collapse when resizing to mobile

### PR DESCRIPTION
**Pull request by rhamilto**
Fixes https://jira.coreos.com/browse/CONSOLE-1020

After:
wlws8e4shk

### CODE REVIEW

File: frontend/public/components/app.jsx [hunk 1]
```diff
@@ -104,12 +104,18 @@ class App extends React.PureComponent {
...
+    this._onResize = this._onResize.bind(this);
+    this.previousDesktopState = this._isDesktop();
 
     this.state = {
       isNavOpen: this._isDesktop(),
     };
   }
 
+  componentWillMount() {
+    window.addEventListener('resize', this._onResize);
```
Comment (spadgett) [inline frontend/public/components/app.jsx]: We should remove the listener on unmount

File: frontend/public/components/app.jsx [hunk 2]
```diff
@@ -104,12 +104,23 @@ class App extends React.PureComponent {
...
+  }
+
+  componentWillUnmount() {
+    super.componentWillUnmount();
```
Comment (spadgett) [inline frontend/public/components/app.jsx]: I don't think we need the call to `super.componentWillUnmount()` since this isn't extending `SafetyFirst`

Comment (rhamilto) [reply frontend/public/components/app.jsx]: Updated

File: frontend/public/style/_overrides.scss [hunk 3]
```diff
@@ -284,11 +284,16 @@ tags-input .autocomplete .suggestion-item em {
...
   --pf-c-page__sidebar--PaddingTop: 0;
   position: relative; // fix z-index bug in Edge
+  width: 0 !important; // only set size when expanded (.pf-m-expanded is added)
 
   @media screen and (min-width: $grid-float-breakpoint) {
     --pf-c-page__sidebar--BoxShadow: none;
   }
 
+  &.pf-m-expanded {
+    width: var(--pf-c-page__sidebar--md--Width) !important; // maintain desktop width so size doesn't animate before hiding when resizing to mobile
```
Comment (spadgett) [inline frontend/public/style/_overrides.scss]: Thanks for fixing that

**Comment by rhamilto** [comment]
Sorry.  I'm a dolt and committed and force pushed in the wrong terminal tab, resulting in this change.  There are no new changes.

**Comment by spadgett** [comment]
The local console couldn't find index.html on startup, hm.

> index.html not found in configured public-dir path: open frontend/public/dist/index.html: no such file or directory

/retest

**Pull request merged**